modules/apps/transfer/ibc_module_test.go

```diff
@@ -273,6 +276,157 @@ func (suite *TransferTestSuite) TestOnChanOpenAck() {
 	}
 }
 
+func (suite *TransferTestSuite) TestOnRecvPacket() {
+	// This test suite mostly covers the top-level logic of the ibc module OnRecvPacket function
+	// The core logic is covered in keeper OnRecvPacket
+	var packet channeltypes.Packet
+	testCases := []struct {
+		name          string
+		malleate      func()
+		expAckSuccess bool
+		expAsyncAck   bool
```

Comment:
should we use `expAck` instead?

Reply:
Yeah, I considered that, and I don't actually remember why I didn't do it. But it sure looks like it might be better here. I'll give it a shot.